Fix tracing warnings with gcc-7

**Describe the bug**
```
/home/pbmille/workspace/src/EMPIRE/TPL/vt/src/vt/trace/trace.cc: In member function ‘vt::trace::TraceEventIDType vt::trace::Trace::logEvent(std::unique_ptr<vt::trace::Log>)’:
/home/pbmille/workspace/src/EMPIRE/TPL/vt/src/vt/trace/trace.cc:667:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       and traces_.size() >= ArgType::vt_trace_flush_size) {
           ~~~~~~~~~~~~~~~^~~~~~~~~~
/home/pbmille/workspace/src/EMPIRE/TPL/vt/src/vt/trace/trace.cc: In member function ‘void vt::trace::Trace::flushTracesFile(bool)’:
/home/pbmille/workspace/src/EMPIRE/TPL/vt/src/vt/trace/trace.cc:722:22: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (traces_.size() >= ArgType::vt_trace_flush_size) {
       ~~~~~~~~~~~~~~~^~~~~~~~~~
```

--- a/src/vt/trace/trace.cc
+++ b/src/vt/trace/trace.cc
@@ -659,7 +659,7 @@ TraceEventIDType Trace::logEvent(LogType&& log) {
   // If auto-flush, can flush immediately.
   // TODO: log time of flushing; unify with group-end.
   if (ArgType::vt_trace_flush_size not_eq 0
-      and traces_.size() >= ArgType::vt_trace_flush_size) {
+      and traces_.size() >= static_cast<std::size_t>(ArgType::vt_trace_flush_size)) {
     writeTracesFile(incremental_flush_mode);
   }
 
@@ -714,7 +714,7 @@ void Trace::flushTracesFile(bool useGlobalSync) {
     // (Consider pushing out: barrier usages are probably domain-specific.)
     theCollective()->barrier();
   }
-  if (traces_.size() >= ArgType::vt_trace_flush_size) {
+  if (traces_.size() >= static_cast<std::size_t>(ArgType::vt_trace_flush_size)) {
     writeTracesFile(incremental_flush_mode);
   }
 }

--- a/src/vt/trace/trace_log.h
+++ b/src/vt/trace/trace_log.h
@@ -85,10 +85,10 @@ struct Log final {
       friend struct Log;
 
       SysData(
-        TraceMsgLenType msg_len,
+        TraceMsgLenType in_msg_len,
         uint64_t in_idx1 = 0, uint64_t in_idx2 = 0,
         uint64_t in_idx3 = 0, uint64_t in_idx4 = 0
-      ) : msg_len(msg_len),
+      ) : msg_len(in_msg_len),
           idx1(in_idx1), idx2(in_idx2), idx3(in_idx3), idx4(in_idx4)
       {
       }
@@ -120,10 +120,10 @@ struct Log final {
       friend struct Log;
 
       UserData(
-        std::string const& user_note, UserDataType user_data,
-        UserEventIDType user_event, bool user_start
-      ) : user_note(user_note), user_data(user_data),
-          user_event(user_event), user_start(user_start)
+        std::string const& in_user_note, UserDataType in_user_data,
+        UserEventIDType in_user_event, bool in_user_start
+      ) : user_note(in_user_note), user_data(in_user_data),
+          user_event(in_user_event), user_start(in_user_start)
       {
       }
 

